ocaml/xapi/repository.ml

```diff
@@ -793,9 +788,17 @@ let apply_updates ~__context ~host ~hash =
         let updates_info =
           parse_updateinfo ~__context ~self:repository ~check:true
         in
-        if Helpers.is_pool_master ~__context ~host then
-          maybe_set_restart_for_all_vms ~__context ~repository_name
-            ~updates_info ;
+        let updates_of_hosts =
+          if Helpers.is_pool_master ~__context ~host then (
+            (* save available updates before applying on coordinator *)
+            if Hashtbl.length updates_in_cache = 0 then
+              set_available_updates ~__context |> ignore ;
+            let hosts = Db.Host.get_all ~__context in
+            consolidate_updates_of_hosts ~repository_name ~updates_info ~hosts
+            |> fst
+          ) else
+            []
+        in
         let host_updates =
```

Comment:
Can we get `host_updates` directly from `updates_of_hosts` if `is_pool_master`?

Reply:
Yes. But it may not be worth do one more special thing here.
(Supporting RestartVM is introducing many special things :( )